merge/ignore_test.go

```diff
@@ -0,0 +1,228 @@
+package merge_test
+
+import (
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/v3/fieldpath"
+	. "sigs.k8s.io/structured-merge-diff/v3/internal/fixture"
+)
+
+func TestIgnoredFields(t *testing.T) {
+	tests := map[string]TestCase{
+		"update_does_not_own_ignored": {
+			APIVersion: "v1",
+			Ops: []Operation{
+				Update{
+					Manager:    "default",
+					APIVersion: "v1",
+					Object: `
+						numeric: 1
+						string: "some string"
+					`,
+					IgnoredFields: fieldpath.NewSet(
+						fieldpath.MakePathOrDie("string"),
+					),
+				},
+				ExpectState{
```

Comment:
See, now we have more code, and you don't even know which feature to use. You technically don't have to use `ExpectState` here, not `ExpectManagedFields`, I don't know if that's needed. You can always split the tests ...

Reply:
Or just remove the other way to check?

Reply:
Also, `ExpectState` is also not the best name, it's actually expecting a specific object?

Reply:
Okay, I remember having this discussion.
I'll split the tests and remove the operations.

But I'll have to add a way for checking managed fields.

Without a way to check intermediate state, I'll have to split and duplicate where necessary. That will be more code, but probably also easier to debug.